module2/src/com/epam/onedimensionalarrays/SumOfIndexesOfPrimeNumbers.java

```diff
@@ -0,0 +1,19 @@
+package com.epam.onedimensionalarrays;
+//Задана последовательность N вещественных чисел.
+//Вычислить сумму чисел, порядковые номера которых являются простыми числами.
+
+public class SumOfIndexesOfPrimeNumbers {
+    public static int sumOfIndexesOfPrimeNumbers(int[] array) {
+        int sum = 0;
+        for (int i = 2; i < array.length; i++) {
+            int dividers = 0;
+            for (int j = 1; j <= i; j++) {
+                if (i % j == 0)
+                    dividers++;
+            }
+            if (dividers <= 2)
+                sum += array[i];
+        }
+        return  sum;
+    }
```

Comment:
This logic is more efficient
```suggestion
    private static final int FIRST_PRIME_INDEX = 2;
    private static final int MIN_DIVISOR_FOR_PRIME_INDEX = 2;

    public static double countSum(double[] generatedNumbers) {
        double sum = 0.0;
        for (int currentIndex = FIRST_PRIME_INDEX; currentIndex < generatedNumbers.length; currentIndex++) {
            boolean isCompositeIndex = false;
            for (int divisor = MIN_DIVISOR_FOR_PRIME_INDEX; divisor <= Math.round(Math.sqrt(currentIndex));
                    divisor++) {
                if (currentIndex % divisor == 0) {
                    isCompositeIndex = true;
                    break;
                }
            }
            if (!isCompositeIndex) {
                sum += generatedNumbers[currentIndex];
            }
        }
        return sum;
    }
```